pkgs/development/libraries/science/math/nccl/default.nix

```diff
@@ -17,16 +15,25 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ which addOpenGLRunpath ];
+  nativeBuildInputs = [
+    which
+    addOpenGLRunpath
+    cuda_nvcc
+  ];
 
-  buildInputs = [ cudatoolkit ];
+  buildInputs = [
+    cuda_cudart
+  ];
 
   preConfigure = ''
     patchShebangs src/collectives/device/gen_rules.sh
+    export NVCC_APPEND_FLAGS+=' --compiler-bindir=${cudatoolkit.cc}/bin'
   '';
 
   makeFlags = [
-    "CUDA_HOME=${cudatoolkit}"
+    "CUDA_HOME=${cuda_nvcc}"
+    "CUDA_LIB=${cuda_cudart}/lib64"
```

Comment:
I'm still waiting for the builds to finish, but I think this works. The variable names come from `common.mk` in the nccl repo

Reply:
(it builds)